src/main/java/ua/tc/marketplace/service/CategoryService.java

```diff
@@ -2,42 +2,102 @@
 
 
 import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import ua.tc.marketplace.exception.model.CategoryCreationException;
+import ua.tc.marketplace.exception.model.CategoryNotFoundException;
+import ua.tc.marketplace.exception.model.MappingException;
 import ua.tc.marketplace.model.entity.Category;
 import ua.tc.marketplace.repository.CategoryRepository;
+import ua.tc.marketplace.model.dto.CategoryDto;
+import ua.tc.marketplace.util.mapper.CategoryMapper;
+import ua.tc.marketplace.util.mapper.ClassificationAttributeMapper;
 
 import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class CategoryService {
-
+    @Autowired
     private final CategoryRepository categoryRepository;
+    @Autowired
+    private final CategoryMapper categoryMapper;
+    @Autowired
+    private final ClassificationAttributeMapper classificationAttributeMapper;
+
+
+
+    public Page<CategoryDto> findAll(Pageable pageable) {
+        try {
+            Page<Category> categories = categoryRepository.findAll(pageable);
+            return categories.map(categoryMapper::convertToDto);
+        } catch (RuntimeException e) {
```

Comment:
I think this `try-catch` block is redundant. I suggest to remove it. Or explain why do you use it here?